### DISCUSSION TITLE
Download mail with IMAP

### DISCUSSION THREAD

**Discussion by Xaeeb** [curl/curl#10076] [Q&A] [answered]
Hi Everyone!

When i try to download an email with IMAP, the output file does not contains the mail body, only the FETCH command.
`curl -v "imaps://imapserver.domain/inbox" -X "FETCH 1 BODY.PEEK[]" -u "username:password" -o dummy.eml`

dummy.eml:
`* 1 FETCH (BODY[] {247333}`

Console output:
`< A003 OK [READ-ONLY] Completed`
`> A004 FETCH 1 BODY.PEEK[]`
`< * 33 FETCH (BODY[] {247333}`
`< Return-Path: <bounce+xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx@xxxx.xxx.yyyy.com>`
`< Received: from xxxxxx.internal (xxxxx.xxxx.internal [xxx.xxx.xxx.xxx])`
....... etc

When i try this with libcurl, the CURLOPT_WRITEFUNCTION callback function called only once with the same result.
`* 1 FETCH (BODY[] {247333}`

When i try this command, the output file contains correctly the whole mail body. Same result with libcurl.
`curl -v "imaps://imapserver.domain/inbox;mailindex=1" -u "username:password" -o dummy.eml`

Console output:
`< A003 OK [READ-ONLY] Completed`
`> A004 FETCH 33 BODY[]`
`< * 1 FETCH (BODY[] {247333}`
`* Found 247333 bytes to download`
..... etc

What's wrong with my first curl command? curl version 7.86.0. I tried this under openSUSE Tumbleweed and Windows 10 MinGW-w64.

Regards,
Xaeeb

**Comment by jay** [comment, marked as answer]
It is or isn't a bug depending on how you look at it. See #536. I put it on the known bugs list in 2016 but it was removed during a reorganization.

**Reply by Xaeeb** [reply]
Hello jay!

Thank you for your answer! I think it is a bug, but i can make a workaround with URL configuration. Only problem is the BODY.PEEK[], because i don't want to touch the mail with FETCH, but i can't use it in URL.

Regards,
Xaeeb

**Reply by dschmeer** [reply]
Hello jay!

I have the exact same problem here: I must download mails without changing the Seen flag, which is not possible because there is no way to specify the PEEK.

Maybe it is a possibility to implement MAILINDEXPEEK= and UIDPEEK= in addition to MAILINDEX= and UID=, or to add an additional parameter PEEK=1.

Kind regards
Dieter
